subsys/net/ip/tcp.c

```diff
@@ -28,7 +28,8 @@ LOG_MODULE_REGISTER(net_tcp, CONFIG_NET_TCP_LOG_LEVEL);
 
 #define ACK_TIMEOUT_MS CONFIG_NET_TCP_ACK_TIMEOUT
 #define ACK_TIMEOUT K_MSEC(ACK_TIMEOUT_MS)
-#define FIN_TIMEOUT_MS MSEC_PER_SEC
+/* The first message is sent at t=0, so CONFIG_NET_TCP_RETRY_COUNT * rto should work */
```

Comment:
With this approach, you're not giving the final retransmission time to receive the response (i. e. we send the final retransmission, and timeout just after). I think we should still use `CONFIG_NET_TCP_RETRY_COUNT + 1` in the formula (maybe use `tcp_retries` to keep it shorter) .

Reply:
Yeah, but from what I see in the code, the original transmission does not count into CONFIG_NET_TCP_RETRY_COUNT, so with CONFIG_NET_TCP_RETRY_COUNT = 3, there'll be 4 transmissions in total (with the final one at 600 ms).

Reply:
The comment aslo needs an update, right?

Reply:
There you go